src/app/(components)/vods.tsx

```diff
@@ -74,7 +98,23 @@ export const VODs = async (props: { username: string }) => {
     }
   ).then((response) => response.json());
 
-  const data = (response as TwitchVodRequest).data;
+  const streams = await fetch(
+    `https://api.twitch.tv/helix/streams?user_id=${twitchUserId}&type=live`,
+    {
+      method: "GET",
+      headers: generateTwitchRequestHeaders(creds),
+      redirect: "follow",
+    }
+  ).then((response) => response.json());
+
+  const streamMap = new Map<string, boolean>();
+  (streams as TwitchStreamRequest).data.forEach((stream) => {
+    streamMap.set(stream.id, true);
+  });
+
+  const data = (response as TwitchVodRequest).data.filter((vod) => {
+    return !streamMap.has(vod.stream_id);
+  });
```

Comment:
haven't set up my env and i'm on my phone but it could look something like this:

```suggestion
const [
  { data: vodData }, 
  { data: streamData }
] = await Promise.all([
  fetch(`https://api.twitch.tv/helix/videos?user_id=${twitchUserId}`, {
    method: "GET",
    headers: generateTwitchRequestHeaders(creds),
    redirect: "follow",
  }).then((response) => response.json()),
  fetch(`https://api.twitch.tv/helix/streams?user_id=${twitchUserId}&type=live`, {
    method: "GET",
    headers: generateTwitchRequestHeaders(creds),
    redirect: "follow",
  }).then((response) => response.json())
]);

const streamMap = new Map<string, boolean>(streamData.map((stream) => [stream.id, true]));

const filteredVodData = vodData.filter(({ stream_id }) => !streamMap.has(stream_id));
```

Reply:
probably fucked something up there so don't copy/paste. on the bus rn lol

Reply:
alr ima test it out

Reply:
840985e676d1950aefbd0875e606122d991e6ab0

Reply:
noice, looks pretty